src/profiles/profileLookup.js

```diff
@@ -0,0 +1,35 @@
+/* @flow */
+
+/**
+ * Look up a user profile by blockstack ID
+ *
+ * @param {Object} name The Blockstack ID of the profile to look up
+ * @param {string} [profileLookupURL=https://core.blockstack.org/v2/users/] The URL
+ * to use for profile lookup 
+ * @returns {Promise} that resolves to a profile object
+ */
+export function lookupProfile(name: string, profileLookupURL: string = 'https://core.blockstack.org/v2/users/') {
```

Comment:
Resolving profiles should be done in javascript like in the browser.

We can move this browser logic to blockstack.js to achieve this:
https://github.com/blockstack/blockstack-browser/blob/master/app/js/utils/profile-utils.js
https://github.com/blockstack/blockstack-browser/blob/master/app/js/utils/zone-utils.js#L36

One reason this needs to happen is that local core api endpoints don't have the `/v2/users` endpoint (unless i'm mistaken) - I believe this is only resolver feature. 

We'd want to use this end point:

```
curl http://localhost:6270/v1/names/user1.id
{"status": "registered", "zonefile": "$ORIGIN user1.id\n$TTL 3600\n_http._tcp URI 10 1 \"https://gaia.blockstack.org/hub/1FmW15z7yj6t2eiWCjHXq34aAUKBsEF4AS/0/profile.json\"\n", "expire_block": 55245, "blockchain": "bitcoin", "last_txid": "04662e7f514d2e23fa782045abe7a60f01a0cb4cf18b943a9730069681d4cc6e", "address": "1FmW15z7yj6t2eiWCjHXq34aAUKBsEF4AS", "zonefile_hash": "f58c767473e9291ea654965d55b12f1591d11550"}
```

Reply:
Yep, that's correct -- you only get the `/v2/users/` endpoint on the resolver.